homeassistant/components/integration/sensor.py

```diff
@@ -107,9 +108,7 @@ def from_name(method_name: str) -> _IntegrationMethod:
         return _NAME_TO_INTEGRATION_METHOD[method_name]()
 
     @abstractmethod
-    def validate_states(
-        self, left: State, right: State
-    ) -> tuple[Decimal, Decimal] | None:
+    def validate_states(self, left: str, right: str) -> tuple[Decimal, Decimal] | None:
```

Comment:
Why the move from `State` to `str` ?

Reply:
It's because we don't have two complete state objects when only the reported time is bumped